apps/glusterfs/app_volume.go

```diff
@@ -207,6 +207,10 @@ func (a *App) VolumeInfo(w http.ResponseWriter, r *http.Request) {
 		} else if err != nil {
 			http.Error(w, err.Error(), http.StatusInternalServerError)
 			return err
+		} else if !entry.Visible() {
+			// treat an invisible entry like it doesn't exist
+			http.Error(w, "Id not found", http.StatusNotFound)
+			return nil
```

Comment:
i think we need to return ErrNotFound

Reply:
and merge with err == ErrNotFound condition?!